spacy_llm/tests/tasks/test_rel.py

```diff
@@ -0,0 +1,104 @@
+from pathlib import Path
+
+import pytest
+import spacy
+from confection import Config
+from pytest import FixtureRequest
+
+from spacy_llm.tasks.rel import RelationItem
+
+from ..compat import has_openai_key
+
+EXAMPLES_DIR = Path(__file__).parent / "examples"
+
+
+@pytest.fixture
+def zeroshot_cfg_string():
+    return """
+    [nlp]
+    lang = "en"
+    pipeline = ["ner", "llm"]
+    batch_size = 128
+
+    [components]
+
+    [components.ner]
+    source = "en_core_web_md"
+
+    [components.llm]
+    factory = "llm"
+
+    [components.llm.task]
+    @llm_tasks = "spacy.REL.v1"
+    labels = "LivesIn,Visits"
+
+    [components.llm.backend]
+    @llm_backends = "spacy.REST.v1"
+    api = "OpenAI"
+    """
+
+
+@pytest.fixture
+def fewshot_cfg_string():
+    return f"""
+    [nlp]
+    lang = "en"
+    pipeline = ["ner", "llm"]
+    batch_size = 128
+
+    [components]
+
+    [components.ner]
+    source = "en_core_web_md"
+
+    [components.llm]
+    factory = "llm"
+
+    [components.llm.task]
+    @llm_tasks = "spacy.REL.v1"
+    labels = "LivesIn,Visits"
+
+    [components.llm.task.examples]
+    @misc = "spacy.FewShotReader.v1"
+    path = {str(EXAMPLES_DIR / "rel_examples.jsonl")}
+
+    [components.llm.backend]
+    @llm_backends = "spacy.REST.v1"
+    api = "OpenAI"
+    """
+
+
+@pytest.fixture
+def task():
+    text = "Joey rents a place in New York City."
+    gold_relations = [RelationItem(dep=0, dest=1, relation="LivesIn")]
+    return text, gold_relations
+
+
+@pytest.mark.skipif(has_openai_key is False, reason="OpenAI API key not available")
+@pytest.mark.parametrize("cfg_string", ["zeroshot_cfg_string", "fewshot_cfg_string"])
+def test_rel_config(cfg_string, request: FixtureRequest):
+    """Simple test to check if the config loads properly given different settings"""
+
+    cfg_string = request.getfixturevalue(cfg_string)
+    orig_config = Config().from_str(cfg_string)
+    nlp = spacy.util.load_model_from_config(orig_config, auto_fill=True)
+    assert nlp.pipe_names == ["ner", "llm"]
+
+
+@pytest.mark.external
+@pytest.mark.skipif(has_openai_key is False, reason="OpenAI API key not available")
+@pytest.mark.parametrize("cfg_string", ["zeroshot_cfg_string", "fewshot_cfg_string"])
+def test_rel_predict(task, cfg_string, request):
+    """Use OpenAI to get REL results.
+    Note that this test may fail randomly, as the LLM's output is unguaranteed to be consistent/predictable
+    """
+    cfg_string = request.getfixturevalue(cfg_string)
+    orig_config = Config().from_str(cfg_string)
+    nlp = spacy.util.load_model_from_config(orig_config, auto_fill=True)
```

Comment:
I suspect this isn't going to work well without the vectors? I thought about doing this initially but wasn't sure there was a good use case, but it might make sense to extend the `assemble` method to accept a `Config` in addition to a config path?

Reply:
Ah you're right, I changed the example but not the test... Good catch.

Reply:
I'll add an option to supply a config.

Reply:
What do you think of this `config_or_path` parameter?

Reply:
I'm not entirely sure that there aren't some potentially weird side effects related to interpolation and a provided config, but it seems okay to use for here for now.

Reply:
I think the method gets a little weird in terms of the `str` option because it's not clear whether it's a config string or a path string.

Reply:
Yeah, this is why I didn't do `Config` just now, because it was getting too messy. You can go ahead with this for now and I'll think a bit about how to make it more like spacy (probably with two separate methods).